pkg/core/database/heavy/transactions.go

```diff
@@ -591,6 +593,45 @@ func (t transaction) FetchBlockHeightSince(sinceUnixTime int64, offset uint64) (
 
 }
 
+func (t transaction) StoreCandidateMessage(cm message.Candidate) error {
+	buf := new(bytes.Buffer)
+
+	if err := message.MarshalCandidate(buf, cm); err != nil {
+		return err
+	}
+
+	key := append(CandidatePrefix, cm.Block.Header.Hash...)
+	t.put(key, buf.Bytes())
+	return nil
+}
+
+func (t transaction) FetchCandidateMessage(hash []byte) (*message.Candidate, error) {
```

Comment:
Not sure why we're going back again to the idea of storing candidate blocks per round in the blockchain database (persistently) instead of keeping them as a cache This has its drawbacks.

Reply:
Mostly for consistency with the rest of the design. Using the DB facilitates simplicity, and reduces concurrency which wasn't entirely necessary, considering that the DB is easy to access for any part of the node. As the DB is ready for concurrent access, there should not be any danger in using it for this purpose.

However, it is only one part of the larger refactor. If there is enough reason to reintroduce concurrency for fetching candidates, we can revert to it if needed.

Reply:
You have many options here different from reusing the blockchain db instance. Please do not overload the blockchain db with intermediate stuff like candidates. 

If your idea was simply to reduce concurrency (no broker loop), let's keep the approach of in-memory candidates store and provide again direct access (in non-rpcbus manner)

Reply:
I was the one suggesting the use of the DB. In general, DB are designed to withstand thousands of transactions per second without breaking a sweat, so I don't see us hitting an overloading scenario by adding tens of transaction (i.e. storing 6-12 candidates at every selection run). This will also help in eventual post-mortem. 

Caching mechanisms will be reintroduced later on with a dedicated PR once we are convinced that all stability issues are solved